### PR TITLE
Optimize inventory wrappers & Registry validation

### PR DESCRIPTION
**Pull request by 116s**
Some stuff here I'm not sure if It's good or necessary to have.

### CODE REVIEW

File: src/main/java/net/minecraftforge/fluids/FluidRegistry.java [hunk 1]
```diff
@@ -272,7 +268,7 @@ public static boolean addBucketForFluid(Fluid fluid) {
      * @return An immutable set containing the fluids
      */
     public static Set<Fluid> getBucketFluids() {
-        return ImmutableSet.copyOf(bucketFluids);
+        return Collections.unmodifiableSet(bucketFluids);
```
Comment (116s) [inline src/main/java/net/minecraftforge/fluids/FluidRegistry.java:271]: This was the change.

File: src/main/java/net/minecraftforge/fluids/FluidRegistry.java [hunk 2]
```diff
@@ -23,6 +23,7 @@
 import net.minecraftforge.fml.common.registry.RegistryDelegate;
 import org.apache.logging.log4j.Level;
 
+import java.util.Collections;
```
Comment (116s) [inline src/main/java/net/minecraftforge/fluids/FluidRegistry.java:26]: WIth this.

File: src/main/java/net/minecraftforge/fluids/FluidRegistry.java (hunk 1 above)
Comment (asbyth) [reply src/main/java/net/minecraftforge/fluids/FluidRegistry.java:271]: Patcher already makes this change, not necessary

Comment (116s) [reply src/main/java/net/minecraftforge/fluids/FluidRegistry.java]: Done

**Pull request merged**